rogue/src/components/magic.rs

```diff
@@ -14,11 +14,52 @@ use serde::{Serialize, Deserialize};
 // ticking each game turn.
 //------------------------------------------------------------------
 
+#[derive(Component, Serialize, Deserialize, Clone)]
+pub struct BlessingOrb {}
+
+const ORBS_NEEDED_FOR_BLESSING: i32 = 4;
+#[derive(Component, ConvertSaveload, Clone)]
+pub struct BlessingOrbBag {
+    pub count: i32
+}
+impl BlessingOrbBag {
+    pub fn enough_orbs_for_blessing(&self) -> bool {
+        self.count >= ORBS_NEEDED_FOR_BLESSING
+    }
+    pub fn cash_in_orbs_for_blessing(&mut self) {
+        self.count = i32::max(0, self.count - ORBS_NEEDED_FOR_BLESSING);
+    }
+}
+
+#[derive(Component, Serialize, Deserialize, Clone)]
+pub struct BlessingSelectionTile {}
+
+#[derive(Component, Serialize, Deserialize, Clone)]
+pub struct OfferedBlessing {}
+
+
+#[derive(PartialEq, Copy, Clone, Serialize, Deserialize)]
+pub enum BlessingSlot {
```

Comment:
Looks like I had the idea to introduce spell slots, though I don't seem to have made much use of it yet.